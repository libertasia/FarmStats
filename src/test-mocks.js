import {MetricType} from "./const";

export const mockStats = [
  {
    "location": "Noora's farm",
    "datetime": "2018-12-31T22:00:00.000Z",
    "value": 5.88,
    "farm_id": "3",
    "sensor_type": "ph"
  },
  {
    "location": "Friman Metsola collective",
    "datetime": "2018-12-31T22:00:00.000Z",
    "value": 5.88,
    "farm_id": "3",
    "sensor_type": "ph"
  },
  {
    "location": "PartialTech Research Farm",
    "datetime": "2018-12-31T22:00:00.000Z",
    "value": 5.88,
    "farm_id": "3",
    "sensor_type": "ph"
  },
  {
    "location": "Organic Ossi's Impact That Lasts plantase",
    "datetime": "2018-12-31T22:00:00.000Z",
    "value": 5.88,
    "farm_id": "3",
    "sensor_type": "ph"
  },
];

export const mockLocations = [
  {
    "farm_id": "1",
    "name": "Noora's farm",
    "isSelected": true,
  },
  {
    "farm_id": "2",
    "name": "Friman Metsola collective",
    "isSelected": true,
  },
  {
    "farm_id": "3",
    "name": "PartialTech Research Farm",
    "isSelected": true,
  },
  {
    "farm_id": "4",
    "name": "Organic Ossi's Impact That Lasts plantase",
    "isSelected": true,
  }
];

export const mockMetrics = [
  {
    type: MetricType.PH,
    isSelected: true,
  },
  {
    type: MetricType.RAINFALL,
    isSelected: true,
  },
  {
    type: MetricType.TEMPERATURE,
    isSelected: true,
  }
];

export const mockMonthlyStats = [
  {
    "farm_id": "1",
    "sensor_type": "temperature",
    "stats": [
      {
        "month": 12,
        "year": 2019,
        "average": -5.3416,
        "median": -5.8,
        "standard_deviation": 5.0244
      },
      {
        "month": 1,
        "year": 2021,
        "average": -6.228,
        "median": -6,
        "standard_deviation": 4.9634
      },
      {
        "month": 9,
        "year": 2019,
        "average": 21.366,
        "median": 22.5,
        "standard_deviation": 4.5688
      },
    ]
  },
  {
    "farm_id": "2",
    "sensor_type": "temperature",
    "stats": [
      {
        "month": 12,
        "year": 2019,
        "average": -5.3416,
        "median": -5.8,
        "standard_deviation": 5.0244
      },
      {
        "month": 1,
        "year": 2021,
        "average": -6.228,
        "median": -6,
        "standard_deviation": 4.9634
      },
      {
        "month": 9,
        "year": 2019,
        "average": 21.366,
        "median": 22.5,
        "standard_deviation": 4.5688
      },
    ]
  },
  {
    "farm_id": "3",
    "sensor_type": "temperature",
    "stats": [
      {
        "month": 12,
        "year": 2019,
        "average": -5.3416,
        "median": -5.8,
        "standard_deviation": 5.0244
      },
      {
        "month": 1,
        "year": 2021,
        "average": -6.228,
        "median": -6,
        "standard_deviation": 4.9634
      },
      {
        "month": 9,
        "year": 2019,
        "average": 21.366,
        "median": 22.5,
        "standard_deviation": 4.5688
      },
    ]
  },
  {
    "farm_id": "4",
    "sensor_type": "temperature",
    "stats": [
      {
        "month": 12,
        "year": 2019,
        "average": -5.3416,
        "median": -5.8,
        "standard_deviation": 5.0244
      },
      {
        "month": 1,
        "year": 2021,
        "average": -6.228,
        "median": -6,
        "standard_deviation": 4.9634
      },
      {
        "month": 9,
        "year": 2019,
        "average": 21.366,
        "median": 22.5,
        "standard_deviation": 4.5688
      },
    ]
  },
];
