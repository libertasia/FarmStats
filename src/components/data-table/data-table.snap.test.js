import React from "react";
import {Provider} from "react-redux";
import {render} from "@testing-library/react";
import configureStore from "redux-mock-store";
import {mockStats} from "../../test-mocks";
import DataTable from "./data-table";

const mockStore = configureStore({});
const store = {
  DATA: {
    farmsStats: mockStats,
  },
};

it(`DataTable renders correctly`, () => {
  const {container} = render(
      <Provider store={mockStore(store)}>
        <DataTable
          data={mockStats}
        />
      </Provider>
  );

  expect(container).toMatchSnapshot();
});
