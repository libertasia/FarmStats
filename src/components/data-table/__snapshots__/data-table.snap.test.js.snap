// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable renders correctly 1`] = `
<div>
  <div
    class="table"
  >
    <table
      class="table__data"
      role="table"
    >
      <caption
        class="visually-hidden"
      >
        Farms Data
      </caption>
      <thead
        class="table__header"
      >
        <tr
          class="table__header-row"
        >
          <th>
            Location
          </th>
          <th>
            Datetime (UTC)
          </th>
          <th>
            Metric
          </th>
          <th>
            Value
          </th>
        </tr>
      </thead>
      <tbody
        class="table__body"
      >
        <tr
          class="table__body-row"
        >
          <td>
            farm 1
          </td>
          <td>
            <time
              datetime="2018-12-31T22:00:00.000Z"
            >
              2018-12-31, 22:00
            </time>
          </td>
          <td>
            ph
          </td>
          <td>
            5.88
          </td>
        </tr>
        <tr
          class="table__body-row"
        >
          <td>
            farm 2
          </td>
          <td>
            <time
              datetime="2018-12-31T22:00:00.000Z"
            >
              2018-12-31, 22:00
            </time>
          </td>
          <td>
            rainfall
          </td>
          <td>
            5.88
          </td>
        </tr>
        <tr
          class="table__body-row"
        >
          <td>
            farm 3
          </td>
          <td>
            <time
              datetime="2018-12-31T22:00:00.000Z"
            >
              2018-12-31, 22:00
            </time>
          </td>
          <td>
            rainfall
          </td>
          <td>
            5.88
          </td>
        </tr>
        <tr
          class="table__body-row"
        >
          <td>
            farm 4
          </td>
          <td>
            <time
              datetime="2018-12-31T22:00:00.000Z"
            >
              2018-12-31, 22:00
            </time>
          </td>
          <td>
            rainfall
          </td>
          <td>
            5.88
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="MuiTablePagination-root table__pagination css-jtlhu6-MuiTablePagination-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-11bfvty-MuiToolbar-root-MuiTablePagination-toolbar"
      >
        <div
          class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
        />
        <p
          class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
          id="mui-2"
        >
          Rows per page:
        </p>
        <div
          class="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="mui-2 mui-1"
            class="MuiTablePagination-select MuiSelect-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
            id="mui-1"
            role="button"
            tabindex="0"
          >
            10
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value="10"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTablePagination-selectIcon MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
        >
          1â€“4 of 4
        </p>
        <div
          class="MuiTablePagination-actions"
        >
          <button
            aria-label="Go to first page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            title="Go to first page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="FirstPageIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
              />
            </svg>
          </button>
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            title="Go to previous page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </button>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            title="Go to next page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </button>
          <button
            aria-label="Go to last page"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            title="Go to last page"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LastPageIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
